MapleServer2/PacketHandlers/Game/RideHandler.cs

```diff
@@ -40,8 +60,22 @@ private static void HandleStartRide(GameSession session, PacketReader packet)
             long mountUid = packet.ReadLong();
             // [46-0s] (UgcPacketHelper.Ugc()) but client doesn't set this data?
 
+            if (type == RideType.UseItem)
+            {
+                if (!session.Player.Inventory.Items.ContainsKey(mountUid))
```

Comment:
I wonder if down the road we should change this to something like `session.Player.Inventory.HasUid(mountUid)` and essentially disable directly working with/modifying the Items, and make people use the Inventory directly